boards/hifive1/src/main.rs

```diff
@@ -118,6 +120,64 @@ impl KernelResources<e310_g002::chip::E310x<'static, E310G002DefaultPeripherals<
     }
 }
 
+/// This is in a separate, inline(never) function so that its stack frame is
+/// removed when this function returns. Otherwise, the stack space used for
+/// these static_inits is wasted.
+#[inline(never)]
+unsafe fn get_peripherals() -> &'static mut E310G002DefaultPeripherals<'static> {
```

Comment:
I merely followed the nomenclature used in almost all other boards. I can rename it, sure, but then should we open an issue to rename that method for other boards that have it? I think this was introduced in PR #2503 . Specifically I think I copied even the comments for this function from the pi pico board main.rs, so I know at minimum that one calls the same method `get_peripherals`

Reply:
I've renamed the function name for this PR. Out of curiosity, and to make sure I'm understanding Rust and the `static_init!`macro, that macro is doing two things-- declaring and initializing variables as static, right? Which is why it should only really be called once?

On a related note, should we add any checks to the function to make sure it is only really ever called once? Maybe a static bool and an assert (can asserts be turned off in Rust?)?

Reply:
> it should only really be called once?

A particular `static_init!()` usage MUST only be used once, because the static memory is unique essentially based on the file and line number where it is declared. Calling a function that contains a `static_init!()` macro or using it in a loop causes the underlying buffer to be aliased, and we end up with multiple mutable references to the same memory, which is bad.

Also in #3239 we added protections, so if the same static init is run multiple times the kernel will panic.